lgc/patch/FragColorExport.cpp

```diff
@@ -557,10 +557,12 @@ void LowerFragColorExport::updateFragColors(CallInst *callInst, ColorExportValue
   const unsigned compIdx = cast<ConstantInt>(callInst->getOperand(1))->getZExtValue();
   Value *output = callInst->getOperand(2);
 
-  auto it = m_resUsage->inOutUsage.outputLocMap.find(location);
-  if (it == m_resUsage->inOutUsage.outputLocMap.end())
+  InOutLocationInfo origLocInfo = {};
+  origLocInfo.location = location;
+  auto locInfoMapIt = m_resUsage->inOutUsage.outputLocInfoMap.find(origLocInfo.u16All);
+  if (locInfoMapIt == m_resUsage->inOutUsage.outputLocInfoMap.end())
     return;
-  unsigned hwColorTarget = it->second;
+  unsigned hwColorTarget = reinterpret_cast<InOutLocationInfo *>(&locInfoMapIt->second)->location;
```

Comment:
Sorry for the delay response. We change the definition of location maps for packing usage. Original <key, value> pair use unsigned to represent the original and new locations. Now, InOutLocationInfo is used for mapping and the location is a bit field of it. So we need the cast.
// Represents the location info of input/output
union InOutLocationInfo {
  struct {
    uint16_t half : 1;      // High half in case of 16-bit attriburtes
    uint16_t component : 2; // The component index
    uint16_t location : 10; // The location
    uint16_t isBuiltIn : 1; // Whether location is actually built-in ID
    uint16_t streamId : 2;  // Output vertex stream ID
  };
  uint16_t u16All;
};

Reply:
Ack.

Reply:
Got it. I will use this way to extract the bit field in later changes.